src/errors.rs

```diff
@@ -1,50 +1,72 @@
-use super::Box;
+use super::{dealloc, Channel};
 use core::fmt;
+use core::marker::PhantomData;
 use core::mem;
+use core::ptr::NonNull;
 
 /// An error returned when trying to send on a closed channel. Returned from
 /// [`Sender::send`] if the corresponding [`Receiver`] has already been dropped.
 ///
 /// The message that could not be sent can be retreived again with [`SendError::into_inner`].
 pub struct SendError<T> {
-    channel_ptr: *mut super::Channel<T>,
+    channel_ptr: NonNull<Channel<T>>,
+    // Required due to the reasons outlined in the nomicon section linked below, as well as in the
+    // next section.
+    // https://doc.rust-lang.org/nomicon/dropck.html
+    _dropck: PhantomData<T>,
 }
 
 unsafe impl<T: Send> Send for SendError<T> {}
+unsafe impl<T: Sync> Sync for SendError<T> {}
 
 impl<T> SendError<T> {
-    pub(crate) const fn new(channel_ptr: *mut super::Channel<T>) -> Self {
-        Self { channel_ptr }
+    /// Safety: by calling this function, the caller semantically transfers ownership of the
```

Comment:
Nitpicky. But this should be in the format `/// # Safety`. Since it's documentation on how the function should be used.